health_check/backends.py

```diff
@@ -43,9 +46,26 @@ def add_error(self, error, cause=None):
             logger.error(str(error))
         self.errors.append(error)
 
-    def pretty_status(self):
+    def add_warning(self, warning, cause=None):
```

Comment:
I am sure you can reuse some code here. I would advice to change the structure of the `errors` list and have a new Error type that includes a severity level. Similar to logs really.

Those methods would be merely helpers then.

Reply:
I don't think I would change the structure of the `errors` list given it's public, and people (like me :P) may be relying on it—I wouldn't want to introduce a breaking change. However, I can easily make the two methods share the same code, so I'll update that.

Reply:
Oh, I think I misunderstood what you meant. Yes, I'll just a new "Warning" type of exception and wrap these in that.

Reply:
Alright, gonna take the kiddo to the museum now, so done for the weekend. But here's my update: given all the changes that have made at this point, I instead just opted to remove the `warnings` list altogether, as it is now functionally identical to the `errors` list and we're instead managing that's behavior and output with flags, which I think is better anyway.